app/models/metric/chargeback_helper.rb

```diff
@@ -22,7 +22,15 @@ def tag_list_with_prefix
     if resource.kind_of?(Container)
       state = resource.vim_performance_state_for_ts(timestamp.to_s)
       image_tag_name = "#{state.image_tag_names}|" if state
-      labels = resource.try(:container_image).try(:docker_labels).try(:collect) { |l| "container_image/label/managed/#{l.name}/#{l.value}" }
+      labels = resource.try(:container_image).try(:docker_labels).try(:collect) do |l|
+        escaped_name = AssignmentMixin.escape(l.name)
+        escaped_value = AssignmentMixin.escape(l.value)
+        [
+          # The assignments in tags can appear the old way as they are, or in base64
+          "container_image/label/managed/#{l.name}/#{l.value}",
+          "container_image/label/managed/#{escaped_name}/#{escaped_value}"
+        ]
```

Comment:
I wonder if this too could be encapsulated in AssignmentMixin.

- AFAICT the list built here [gets passed to `AssignmentMixin#get_assigned_for_target`](https://github.com/zeari/manageiq/blob/5ff58f352d7/app/models/chargeback/rates_cache.rb#L23-L24),
  but I don't see an easy non-invasive way move this responsibility inside `get_assigned_for_target`.

- How about just extracting the code generating the escaped tag name from `AssignmentMixin#assign_to_labels` to a helper you can also call here?
 - And possibly add a helper generating the old unescaped tag name, or a helper generating this list of unescaped, escaped tag names?